src/eko/anomalous_dimensions/__init__.py

```diff
@@ -100,17 +101,37 @@ def gamma_ns(order, mode, n, nf):
         eko.anomalous_dimensions.lo.gamma_ns_0 : :math:`\gamma_{ns}^{(0)}(N)`
         eko.anomalous_dimensions.nlo.gamma_nsp_1 : :math:`\gamma_{ns,+}^{(1)}(N)`
         eko.anomalous_dimensions.nlo.gamma_nsm_1 : :math:`\gamma_{ns,-}^{(1)}(N)`
+        eko.anomalous_dimensions.nnlo.gamma_nsp_2 : :math:`\gamma_{ns,+}^{(2)}(N)`
+        eko.anomalous_dimensions.nnlo.gamma_nsm_2 : :math:`\gamma_{ns,-}^{(2)}(N)`
+        eko.anomalous_dimensions.nnlo.gamma_nsv_2 : :math:`\gamma_{ns,v}^{(2)}(N)`
     """
     # cache the s-es
-    s1 = S1(n)
+    sx = np.full(1, harmonics.harmonic_S1(n))
+    # now combine
     gamma_ns = np.zeros(order + 1, np.complex_)
-    gamma_ns[0] = lo.gamma_ns_0(n, s1)
-    if order > 0:
+    gamma_ns[0] = lo.gamma_ns_0(n, sx[0])
+    # NLO and beyond
+    if order >= 1:
+        # TODO: pass the necessary harmonics to nlo gammas
         if mode == "p":
             gamma_ns_1 = nlo.gamma_nsp_1(n, nf)
         elif mode == "m":
             gamma_ns_1 = nlo.gamma_nsm_1(n, nf)
+        # To fill the full valence vector in NNLO we need to add gamma_ns^1 explicitly here
+        elif mode == "v":
+            gamma_ns_1 = nlo.gamma_nsm_1(n, nf)
```

Comment:
Then, instead of duplicating the line of code, simply change the former condition to:
```py
elif mode in ["m", "v"]:
```
In this way it's explicit that the two things are the same, and it's not happening by chance (and you save a couple of lines, nothing major but the result is more readable).